### PR TITLE
drupal-address version 8.x-1.9 build and validation

### PR DESCRIPTION
**Pull request by ambujkumar3**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

**Comment by seth-priya** [comment]
@prashantkhoje please review

File: d/drupal-address/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+How to run drupal related modules test cases.
+-------------
+
+Summary :-
+
+    To run test cases in drupal module we need drupal core package and drupal complete framework which incluse one database,apache server,and core package itself.
+    There are 3 type of tests in drupal unit,functional,intergration. For unit test we dont need drupal full framework like database.
+    Unit test does not use database testing.
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:8]:
```suggestion
    To run test cases in drupal module, we need drupal core package and drupal complete framework which includes a database, apache server and core package itself.
    There are 3 type of tests in drupal unit, functional, integration. For unit tests, we don't need drupal full framework like database.
    Unit tests don't use database.
```

File: d/drupal-address/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+*************************
+
+Copy following files into VM thats needed to run docker file succesfully.
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:12]:
```suggestion
Copy following files into the VM that is needed to run docker file successfully.
```

File: d/drupal-address/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+     #docker build -t drupal_image .
+     #docker images
+
+
+Then run a container using that image.
+
+    #docker run -it -d drupal_image /bin/bash
+    #docker ps
+    #docker exec -it 69cabf536f6f /bin/bash
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:36]:
```suggestion
    # docker exec -it <container-id> /bin/bash
```

File: d/drupal-address/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+Live drupal webserver will be available at http://<ip>:8081
+
+After that we need to run below command inside container.Run it in single command . Step 1 will take sometime to exec.
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:40]:
```suggestion
After that we need to run below command inside the container. Run it in single command. Step 1 will take some time to exec.
```

File: d/drupal-address/README.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+
+        bash-4.4# pwd
+        /opt/app-root/src/drupal
+	bash-4.4# ./vendor/bin/drush en address
+           [notice] Already enabled: address
+
+    cd /opt/app-root/src/drupal/core
+    bash-4.4# pwd
+    /opt/app-root/src/drupal/core
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:78]:
```suggestion
    # cd /opt/app-root/src/drupal

    # yum install -y git php php-gd
    # composer require 'drupal/token:^1.10'
    # composer require 'commerceguys/addressing:*'
    # cd /opt/app-root/src/drupal
	# ./vendor/bin/drush en address
           [notice] Already enabled: address

    # cd /opt/app-root/src/drupal/core
```

File: d/drupal-address/README.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+
+Go to :-
+
+    #cd /opt/app-root/src/drupal/modules
+
+Clone the module which you wanted to test :-
+
+    #git clone https://git.drupalcode.org/project/address.git
+    #git checkout <versions>
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:58]:
```suggestion
    # git clone https://git.drupalcode.org/project/address.git
    # git checkout <versions>
```

File: d/drupal-address/README.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    su - postgres -c 'if [[ $(psql -l | grep dru2_pg) ]]; then     echo "Database already configured..."; else    /usr/pgsql-13/bin/pg_ctl -D /var/lib/pgsql/13/data/ start;     /usr/pgsql-13/bin/createdb -T template0 dru2_pg;     psql dru2_pg < /opt/app-root/src/drupal_schema.sql; fi'
+
+Step 2:-
+
+    /usr/sbin/httpd -k start;
+
+
+Go to :-
+
+    #cd /opt/app-root/src/drupal/modules
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:53]:
```suggestion
    # cd /opt/app-root/src/drupal/modules
```

File: d/drupal-address/README.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+Now create an image from dockerfile (Dockerfile.drupal.ubi i.e Dockerfile)
+
+     #docker build -t drupal_image .
+     #docker images
+
+
+Then run a container using that image.
+
+    #docker run -it -d drupal_image /bin/bash
+    #docker ps
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:35]:
```suggestion
    # docker run -it -d drupal_image /bin/bash
    # docker ps
```

File: d/drupal-address/README.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    #cp Dockerfile.drupal.ubi Dockerfile
+    #cp automate_drupal.sh.txt automate_drupal.sh
+    #chmod +x automate_drupal.sh
+
+
+Now create an image from dockerfile (Dockerfile.drupal.ubi i.e Dockerfile)
+
+     #docker build -t drupal_image .
+     #docker images
+
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:30]:
```suggestion
     # docker build -t drupal_image .
     # docker images
```

File: d/drupal-address/README.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    1) automate_drupal.sh.txt
+    2) Dockerfile.drupal.ubi
+    3) drupal.zip
+    4) drupal_schema.sql
+
+Rename 2 files:-
+
+    #cp Dockerfile.drupal.ubi Dockerfile
+    #cp automate_drupal.sh.txt automate_drupal.sh
+    #chmod +x automate_drupal.sh
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:23]:
```suggestion
    # cp Dockerfile.drupal.ubi Dockerfile
    # cp automate_drupal.sh.txt automate_drupal.sh
    # chmod +x automate_drupal.sh
```

File: d/drupal-address/README.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+    cd /opt/app-root/src/drupal/core
+    bash-4.4# pwd
+    /opt/app-root/src/drupal/core
+
+
+RUN TEST:-
+----------
+
+    bash-4.4# ../vendor/phpunit/phpunit/phpunit ../modules/address/tests/src
+
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:85]:
```suggestion
    # ../vendor/phpunit/phpunit/phpunit ../modules/address/tests/src
```

File: d/drupal-address/README.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+Test output
+----------------
+bash-4.4# ../vendor/phpunit/phpunit/phpunit ../modules/address/tests/src
```
Comment (prashantkhoje) [inline d/drupal-address/README.md:89]:
```suggestion
    # ../vendor/phpunit/phpunit/phpunit ../modules/address/tests/src
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+# ----------------------------------------------------------------------------
+# Package          : address
+# Version          : 8.x-1.9
+# Source repo      : https://git.drupalcode.org/project/address.git
+# Tested on        : UBI 8.5
+# Language         : PHP
+# Travis-Check     : False
+# Script License   : Apache License, Version 2 or later
+# Maintainer       : Ambuj Kumar  <Ambuj.Kumar3@ibm.com>
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:11]:
```suggestion
# Maintainer       : Ambuj Kumar <Ambuj.Kumar3@ibm.com>
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+# ----------------------------------------------------------------------------
+
+# Variables
+PACKAGE_NAME=address
+CORE_PACKAGE_NAME=drupal
+PACKAGE_URL=https://git.drupalcode.org/project/address.git
+CORE_PACKAGE_URL=https://github.com/drupal/drupal
+
+PACKAGE_VERSION=${1:-8.x-1.9}
+
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:28]:
```suggestion
PACKAGE_VERSION=${1:-8.x-1.9}
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+yum install -y git php php-dom php-mbstring zip unzip gd gd-devel php-gd php-pdo php-mysqlnd
+
+OS_NAME=`cat /etc/os-release | grep PRETTY_NAME | cut -d '=' -f2 | tr -d '"'`
+
+#Check if package exists
+if [ -d "$CORE_PACKAGE_NAME" ] ; then
+  rm -rf $CORE_PACKAGE_NAME
+  echo "$CORE_PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package if any"
+
+fi
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:39]:
```suggestion
# Check if package exists
if [ -d "$CORE_PACKAGE_NAME" ] ; then
    rm -rf $CORE_PACKAGE_NAME
    echo "$CORE_PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package"
fi
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+if ! git clone $CORE_PACKAGE_URL $CORE_PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$CORE_PACKAGE_URL $CORE_PACKAGE_NAME"
+        echo "$CORE_PACKAGE_NAME  |  $CORE_PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 0
+fi
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:46]:
```suggestion
if ! git clone $CORE_PACKAGE_URL $CORE_PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$CORE_PACKAGE_URL $CORE_PACKAGE_NAME"
    echo "$CORE_PACKAGE_NAME  |  $CORE_PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 1
fi
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+composer require 'commerceguys/addressing:*'
+
+
+
+if ! composer install --no-interaction; then
+        echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
+        exit 1
+fi
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:60]:
```suggestion

if ! composer install --no-interaction; then
    echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
    exit 1
fi
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+
+    cd modules/
+
+ if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 0
+fi
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:69]:
```suggestion
cd modules/

if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 1
fi
```

File: d/drupal-address/drupal-address_ubi_8.5.sh [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+        echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails"
+        exit 1
+else
+        echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success"
+        exit 0
+fi
```
Comment (prashantkhoje) [inline d/drupal-address/drupal-address_ubi_8.5.sh:89]:
```suggestion
if ! ../vendor/phpunit/phpunit/phpunit ../modules/address/tests/src; then
    echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails"
    exit 1
else
    echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success"
    exit 0
fi
```

File: d/drupal-address/README.md (hunk 1 above)
Comment (ambujkumar3) [reply d/drupal-address/README.md:8]: done the changes according to review comment

**Comment by prashantkhoje** [comment]
LGTM

**Pull request merged**